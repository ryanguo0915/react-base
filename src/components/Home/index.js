import React  from "react";
import EarthquakeTable from '../EarthquakeTable/EarthquakeTable'
import "./index.css";

export default function Home({data}) {
    const {metadata, features} = data;
  return (
    <div className="home__wrapper">
        <h3 className="home__title">{metadata.title}</h3>
        <EarthquakeTable features={features}/>
    </div>
  );
}