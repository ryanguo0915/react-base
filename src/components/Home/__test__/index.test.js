import React from 'react';
import { shallow } from 'enzyme';
import Home from '../index';
describe("Home", () => {
  it("should render my component", () => {
    const props = { 
        data: {
            metadata: { title: 'Title' },
            features: [{
                id:'1', 
                properties:{title:'title', mag: 'mag', time: 'time', status: 'status', tsunami: 'tsunami', type: 'type'}
            }], 
        }
    };
    const wrapper = shallow(<Home {...props}/>);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});