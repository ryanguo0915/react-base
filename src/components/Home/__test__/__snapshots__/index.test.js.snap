// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render my component 1`] = `
Array [
  <div
    className="home__wrapper"
  >
    <h3
      className="home__title"
    >
      Title
    </h3>
    <EarthquakeTable
      features={
        Array [
          Object {
            "id": "1",
            "properties": Object {
              "mag": "mag",
              "status": "status",
              "time": "time",
              "title": "title",
              "tsunami": "tsunami",
              "type": "type",
            },
          },
        ]
      }
    />
  </div>,
]
`;
